core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java

```diff
@@ -126,23 +197,681 @@ public Response getInfo(@QueryParam(QUERY_RAW_CONFIGURATION) final Boolean rawCo
+          .setLiveWorkerNodes(mBlockMaster.getWorkerCount())
+          .setMasterNodeAddress(mMasterProcess.getRpcAddress().toString())
+          .setStartTime(CommonUtils.convertMsToDate(mMetaMaster.getStartTimeMs()))
+          .setStorageTierInfos(infos).setUptime(CommonUtils
+              .convertMsToClockTime(System.currentTimeMillis() - mMetaMaster.getStartTimeMs()))
+          .setUsedCapacity(FormatUtils.getSizeFromBytes(mBlockMaster.getUsedBytes()))
+          .setVersion(RuntimeConstants.VERSION);
+    });
+  }
+
+  /**
+   * Get the information required for the UI's browse page
+   * @param requestPath
+   * @param requestOffset
+   * @param requestEnd
+   * @param requestLimit
+   * @return
+   */
+  @GET
+  @Path(WEBUI_BROWSE)
+  @ReturnType("alluxio.wire.WebUIBrowse")
+  public Response getWebUIBrowse(@DefaultValue("/") @QueryParam("path") String requestPath,
+      @DefaultValue("0") @QueryParam("offset") String requestOffset,
+      @DefaultValue("") @QueryParam("end") String requestEnd,
+      @DefaultValue("20") @QueryParam("limit") String requestLimit) {
+    return RestUtils.call(() -> {
+      if (!Configuration.getBoolean(PropertyKey.WEB_FILE_INFO_ENABLED)) {
+        return new WebUIBrowse();
+      }
+
+      if (SecurityUtils.isSecurityEnabled() && AuthenticatedClientUser.get() == null) {
+        AuthenticatedClientUser.set(LoginUser.get().getName());
+      }
+
+      AlluxioURI currentPath = new AlluxioURI(requestPath);
+      List<FileInfo> filesInfo = null;
+      List<UIFileBlockInfo> fileBlocks = null;
+      long offset = 0;
+      long viewingOffset = offset;
+      String accessControlException = "";
+      String blockSizeBytes = "";
+      String fatalError = "";
+      String fileData = "";
+      String fileDoesNotExistException = "";
+      String highestTierAlias = "";
+      String invalidPathError = "";
+      String invalidPathException = "";
+      UIFileInfo currentDirectory = null;
+      UIFileInfo[] pathInfos = null;
+
+      try {
+        long fileId = mFileSystemMaster.getFileId(currentPath);
+        FileInfo fileInfo = mFileSystemMaster.getFileInfo(fileId);
+        UIFileInfo currentFileInfo = new UIFileInfo(fileInfo);
+        if (currentFileInfo.getAbsolutePath() == null) {
+          throw new FileDoesNotExistException(currentPath.toString());
+        }
+        currentDirectory = currentFileInfo;
+        blockSizeBytes = currentFileInfo.getBlockSizeBytes();
+        if (!currentFileInfo.getIsDirectory()) {
+          long relativeOffset = 0;
+          try {
+            relativeOffset = Long.parseLong(requestOffset);
+          } catch (NumberFormatException e) {
+            relativeOffset = 0;
+          }
+          String endParam = requestEnd;
+          // If no param "end" presents, the offset is relative to the beginning; otherwise, it is
+          // relative to the end of the file.
+          if (endParam == null) {
```

Comment:
This should be `isEmpty` since the default value of `endParam` is `""`

Reply:
I switched to not having it be a defaulted param

Reply:
I slightly prefer having it default to empty, its nice to guarantee all the fields are not null

Reply:
okay, set a default for the parameter and will check for equality to "" instead of nullness